src/distilabel/llm/ollama.py

```diff
@@ -116,22 +163,23 @@ def model_name(self) -> str:
         return self.model
 
     def _api_available(self):
-        """Calls GET {OLLAMA_HOST}"""
-        msg = f"Could not connect to Ollama as {self.OLLAMA_HOST}. Check https://github.com/ollama/ollama for deployment guide."
+        """Checks if the Ollama API is available."""
+        msg = f"Could not connect to Ollama at {self.OLLAMA_HOST}. Check https://github.com/ollama/ollama-python/tree/main for deployment guide."
         try:
-            response = request.urlopen(self.OLLAMA_HOST)
-            if response.getcode() != 200:
-                raise Exception
-        except Exception as e:
+            ollama.list()
+        except ollama.ResponseError as e:
             raise ValueError(msg) from e
 
     def _api_model_available(self):
+        """Checks if the Ollama model is available"""
         msg = f"Model {self.model} is not available. Run `ollama run {self.model}` to serve the model."
         try:
-            self._text_generation_with_backoff(
-                prompt=[{"role": "user", "content": "hi"}], max_tokens=1
+            ollama.chat(
+                model=self.model,
+                messages=[{"role": "user", "content": "hi"}],
+                options={"num_predict": 1},
```

Comment:
Isn't this a bit more time consuming than calling `ollama.list` and then checking whether the Ollama model is listed there?